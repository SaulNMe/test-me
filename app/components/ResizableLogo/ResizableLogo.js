import React, { PureComponent } from 'react';
import { Image  } from 'react-native';
import PropTypes from 'prop-types';

import styles from './ResizableLogoStyle';

export default class ResizableLogo extends PureComponent {
	render() {
		let imageSource;
		switch(this.props.size) {
			case "small":
				imageSource =  require('qualificame-native/assets/small_logo.png');
			break;
			case "medium":
				imageSource = require('qualificame-native/assets/medium_logo.png');
			break;
			case "large":
				imageSource =  require('qualificame-native/assets/large_logo.png');
			break;
		}
	
		return (
			<React.Fragment>
				<Image
					source={imageSource}
				/>
			</React.Fragment>
		);
	}
}

	ResizableLogo.propTypes = {
		size: PropTypes.string
	}

	ResizableLogo.defaultProps = {
		size: "small"
	}
