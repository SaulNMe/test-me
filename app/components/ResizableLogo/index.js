import ResizableLogo from './ResizableLogo';

export default ResizableLogo;
