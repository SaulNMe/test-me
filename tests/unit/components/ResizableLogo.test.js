import React from 'react';
import { shallow } from 'enzyme';
import { Image } from 'react-native';
import ResizableLogo from 'qualificame-native/app/components/ResizableLogo';

function createTestProps (props) {
	return {
		// common props
		size: "medium",
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<ResizableLogo {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays an image', () => {
		expect( wrapper.find(Image).exists()).toBeTruthy();
	});

	describe('with no props', () => {
		beforeEach(()=>{
			wrapper = createWrapper();
		});

		it('displays a default image', () => {
			expect( wrapper.find(Image).exists()).toBeTruthy();
		});
	});
});
