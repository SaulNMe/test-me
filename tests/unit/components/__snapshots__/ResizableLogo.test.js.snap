// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Fragment>
  <Component
    source={1}
  />
</Fragment>
`;
